stdlib/stdlib.mli

```diff
@@ -41,6 +41,13 @@ val invalid_arg : string -> 'a
 val failwith : string -> 'a
 (** Raise exception [Failure] with the given string. *)
 
+val todo : ?msg:string -> ?__LOC__:string -> unit -> _
+[@@alert todo "TODO: not implemented yet"]
+(** Raise an exception [Failure]
+    with provided optional location [?__LOC__] and optional message [?msg].
+    @since 5.3
+*)
```

Comment:
Correct me if I’m wrong, but to me, the main benefit of this function is to get compile-time warnings. If that’s the case, then we should be able to test that compiler output in the testsuite, too.

Reply:
i feel like the main benefit is to have a shorthand to signify that we will implement a given piece of functionality later.
Also it should be possible to get a compile time Error from this: 
```ocaml
[@@@ocaml.alert "++todo"]
let () = 
 Stdlib.todo () 
```
should give us: 
```bash
Line 3, characters 0-11:
3 | Stdlib.todo ();;
    ^^^^^^^^^^^
Error (alert todo): Stdlib.todo
TODO: not implemented yet
```
I will push the test for the error.
Or you thought about introducing a new kind of warning into the compiler?